src/ui/components/Timeline/LoadingProgressBars.tsx

```diff
@@ -0,0 +1,27 @@
+import { clamp } from "lodash";
+import { ReactElement } from "react";
+import { useSelector } from "react-redux";
+import { getLoadedRegions } from "ui/reducers/app";
+import { getZoomRegion } from "ui/reducers/timeline";
+import { overlap } from "ui/utils/timeline";
+
+import styles from "./LoadingProgressBars.module.css";
+
+export default function LoadingProgressBars() {
+  const loadedRegions = useSelector(getLoadedRegions);
+  const zoomRegion = useSelector(getZoomRegion);
+
+  const regions = loadedRegions ? overlap(loadedRegions.indexed, loadedRegions.loaded) : [];
+  const endTime = zoomRegion ? zoomRegion.endTime : 0;
+
+  return regions.map(({ begin, end }, index) => (
+    <div
+      key={index}
+      className={styles.Region}
+      style={{
+        left: `${(begin.time / endTime) * 100}%`,
+        width: `${clamp(((end.time - begin.time) / endTime) * 100, 0, 100)}%`,
+      }}
+    />
+  )) as any;
```

Comment:
Why is TypeScript complaining about the return type of this map?

```
src/ui/components/Timeline/index.tsx:125:38 - error TS2786: 'LoadingProgressBars' cannot be used as a JSX component.
  Its return type 'Element[]' is not a valid JSX element.
    Type 'Element[]' is missing the following properties from type 'ReactElement<any, any>': type, props, key

125             {showLoadingProgress && <LoadingProgressBars />}
                                         ~~~~~~~~~~~~~~~~~~~
```

This error means nothing to me.

Reply:
This is a known weakness in the React TS types.

As _you_ well know, it's been legal to return an array from rendering ever since React 16.

But, the TS types somehow struggle with this.

The solution is to wrap the array in a seemingly-unnecessary fragment to make TS happy.

Reply:
Goofy.